src/test/java/com/iexec/sms/tee/config/TeeWorkerPipelineConfigurationTests.java

```diff
@@ -0,0 +1,165 @@
+        assertEquals(PRE_FINGERPRINT, preComputeConfig.fingerprint());
+        assertEquals(PRE_HEAP_SIZE, preComputeConfig.heapSize());
+        assertEquals(PRE_ENTRYPOINT, preComputeConfig.entrypoint());
+
+        final TeeWorkerPipelineConfiguration.StageConfig postComputeConfig = firstPipeline.getPostCompute();
+        assertEquals(POST_IMAGE, postComputeConfig.image());
+        assertEquals(POST_FINGERPRINT, postComputeConfig.fingerprint());
+        assertEquals(POST_HEAP_SIZE, postComputeConfig.heapSize());
+        assertEquals(POST_ENTRYPOINT, postComputeConfig.entrypoint());
+    }
+
+    @Test
+    void shouldFailWhenPipelineConfigIsNull() {
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration>> violations = validator.validate(new TeeWorkerPipelineConfiguration(null));
+        assertFalse(violations.isEmpty());
+        assertThat(violations)
+                .extracting(ConstraintViolation::getMessage)
+                .containsExactly("Pipeline list must not be empty");
+    }
+
+    @Test
+    void shouldFailIfEmptyPipelineList() {
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration>> violations = validator.validate(new TeeWorkerPipelineConfiguration(Collections.emptyList()));
+        assertFalse(violations.isEmpty());
+        assertThat(violations)
+                .extracting(ConstraintViolation::getMessage)
+                .containsExactly("Pipeline list must not be empty");
+    }
+
+    @Test
+    void shouldFailWhenListElementsAreNull() {
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration>> violations = validator.validate(
+                new TeeWorkerPipelineConfiguration(Collections.singletonList(null)));
+        assertFalse(violations.isEmpty());
+        assertThat(violations)
+                .extracting(ConstraintViolation::getMessage)
+                .containsExactly("List elements must not be null");
+    }
+
+    @Test
+    void shouldFailIfBlankVersion() {
+        final TeeWorkerPipelineConfiguration.Pipeline pipeline = new TeeWorkerPipelineConfiguration.Pipeline(
+                "",
+                preCompute,
+                postCompute
+        );
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration.Pipeline>> violations = validator.validate(pipeline);
+        assertFalse(violations.isEmpty());
+        assertThat(violations)
+                .extracting(ConstraintViolation::getMessage)
+                .containsExactly("Pipeline version must not be blank");
+    }
+
+    @Test
+    void shouldFailIfNullStageConfig() {
+        final TeeWorkerPipelineConfiguration.Pipeline pipeline = new TeeWorkerPipelineConfiguration.Pipeline(
+                VERSION,
+                null,
+                null
+        );
+        final Set<ConstraintViolation<TeeWorkerPipelineConfiguration.Pipeline>> violations = validator.validate(pipeline);
+        assertEquals(2, violations.size());
```

Comment:
Alternate possibility
```suggestion
        assertThat(violations).hasSize(2);
```